### ISSUE TITLE
Cosmos + System.Text.Json throw Newtonsoft.Json exception

### ISSUE THREAD

**Issue by uyen-luu** [Azure/azure-cosmos-dotnet-v3#4942]
>We are continuously addressing and improving the SDK, if possible, make sure the problem persist in the [latest SDK version](https://www.nuget.org/packages/Microsoft.Azure.Cosmos).

**Describe the bug**
Configuring CosmosClient with UseSystemTextJsonSerializerWithOptions works in v3.45.0 but it throws "System.IO.FileLoadException: Could not load file or assembly 'Newtonsoft.Json, Version=10.0.0.0" in the v3.46.0

**To Reproduce**

Configuring the Cosmos Client like this by using  Microsoft.Azure.Cosmos@3.45.0 and check that the CosmosClient has been initialized properly

> var cosmosOptions = new CosmosClientOptions()
> {
>     UseSystemTextJsonSerializerWithOptions = new()
>     {
>         PropertyNamingPolicy = JsonNamingPolicy.CamelCase,
>         DefaultIgnoreCondition = JsonIgnoreCondition.WhenWritingNull,
>         WriteIndented = true,
>     },
>     MaxRetryWaitTimeOnRateLimitedRequests = TimeSpan.FromMinutes(1), // increased from 30 seconds to 60 seconds
>     AllowBulkExecution = false,
> };
> 
> return new CosmosClient(config.ConnectionString, cosmosOptions);

Upgrading to Microsoft.Azure.Cosmos@3.46.0 and re-launch the application.

**Expected behavior**
It should works properly like what it had done in v 3.45.0

**Actual behavior**

Exception raised:
`Could not load file or assembly 'Newtonsoft.Json, Version=10.0.0.0, Culture=neutral, PublicKeyToken=30ad4fe6b2a6aeed'. The system cannot find the file specified./`

**Environment summary**
SDK Version: 3.46.0
OS Version: Windows

**Comment by Pilchie** [comment]
Yes, it is still needed by some internal parts of Cosmos DB.  You should get a build error when you build if you don't have a reference to Newtonsoft.Json.  See https://github.com/Azure/azure-cosmos-dotnet-v3/pull/4839 for details.
